Fix normative notes and clarify/simplify robustness rules

joeyparrish opened the pull request:
Closes #521


<!--
    This comment and the below content is programmatically generated.
    You may add a comma-separated list of anchors you'd like a
    direct link to below (e.g. #idl-serializers, #idl-sequence):

    Don't remove this comment or modify anything below this line.
    If you don't want a preview generated for this pull request,
    just replace the whole of this comment's content by "no preview"
    and remove what's below.
-->
***
<a href="https://pr-preview.s3.amazonaws.com/w3c/encrypted-media/pull/549.html" title="Last updated on May 7, 2026, 9:36 PM UTC (5249374)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/encrypted-media/549/9b445db...5249374.html" title="Last updated on May 7, 2026, 9:36 PM UTC (5249374)">Diff</a>

joeyparrish committed:
Fix normative notes and clarify/simplify robustness rules

Issue #521

marcoscaceres commented on encrypted-media-respec.html:
```diff
@@ -2544,20 +2544,19 @@ <h2>
                 The robustness level associated with the content type. The empty string indicates
                 that any ability to decrypt and decode the content type is acceptable.
+                Implementations MUST configure the [=CDM=] to support at least the robustness
+                levels specified in the configuration of the {{MediaKeySystemAccess}} object used
+                to create the {{MediaKeys}} object.
```
```suggestion
                levels specified in the configuration of the {{MediaKeySystemAccess}} dictionary used
                to create the {{MediaKeys}} instance.
```

chrisn replied:
I think "MediaKeySystemAccess object" is right here - it's an interface, whereas MediaKeySystemConfiguration is a dictionary.

marcoscaceres replied:
You're right, @chrisn — `MediaKeySystemAccess` is an interface, not a dictionary. I was confusing it with `MediaKeySystemConfiguration`. Withdrawing my suggestion.

marcoscaceres commented on encrypted-media-respec.html:
```diff
@@ -2544,20 +2544,19 @@ <h2>
                 <p>
-                  Applications SHOULD specify the robustness level(s) they require to avoid
-                  unexpected client incompatibilities.
+                  Applications are recommended to specify the robustness level(s) they require,
```
```suggestion
                  Developers are encouraged to declare the robustness level(s) they require,
```

marcoscaceres commented on encrypted-media-respec.html:
```diff
@@ -2544,20 +2544,19 @@ <h2>
                 <p>
-                  Applications SHOULD specify the robustness level(s) they require to avoid
-                  unexpected client incompatibilities.
+                  Applications are recommended to specify the robustness level(s) they require,
+                  if any, to avoid unexpected client incompatibilities.
```
```suggestion
                  To avoid unexpected interoperability issues, developers are encouraged to provide the robustness level(s) 
                  (if any).
```

joeyparrish replied:
Dropping this paragraph entirely per the discussion in the other thread, so this suggestion is moot.

marcoscaceres reviewed (commented):
Couple of wording suggestions... I'm not very familiar with how EME works, so would need someone else to check the technical details.

chrisn reviewed (commented):
Sorry for taking so long to respond. I left one comment for consideration.

I notice there's a few other Notes that contain normative statements that we'll need to review too.

marcoscaceres commented on encrypted-media-respec.html:
```diff
@@ -2544,20 +2544,19 @@ <h2>
-                  highest robustness level in the configuration even if a higher robustness level
-                  is available. If only the empty string is specified, implementations MAY be
-                  configured to use the lowest robustness level the implementation supports.
+                  Exact configuration of the [=CDM=] is implementation-specific.
```
I agree with removing the note entirely. The MUST covers the normative requirement, and "Exact configuration of the CDM is implementation-specific" is just restating what MAY already means. The developer guidance ("specify robustness levels to avoid incompatibilities") could go in a non-normative paragraph if @joeyparrish feels it's worth keeping, but I wouldn't object to dropping it.

joeyparrish replied:
Agreed.  I'm removing the note entirely, including the developer-guidance paragraph.

joeyparrish committed:
Merge branch 'main' into normative-robustness

joeyparrish committed:
Remove redundant robustness note

The note's content was either restating what the surrounding normative
prose already conveys or duplicating the meaning of the empty-string
definition. Per review consensus, drop the note entirely.
